components/google_sheets/actions/add-single-row/add-single-row.mjs

```diff
@@ -32,15 +66,64 @@ export default {
         }),
       ],
     },
-    cells: {
-      propDefinition: [
-        googleSheets,
-        "cells",
-      ],
-    },
+    dataFormat,
+  },
+  async additionalProps() {
+    if (this.dataFormat == "column") {
+      if (!this.header) {
+        const rv = {};
+        rv.header = headerProp;
+        return rv;
+      } else {
+        if (this.header === "hasHeaders") {
+          const rv = {};
+          rv.header = headerProp;
+          const { values } = await this.googleSheets.getSpreadsheetValues(this.sheetId, `${this.sheetName}!1:1`);
+          for (let i = 0; i < values[0].length; i++) {
+            rv[`col_${i.toString().padStart(4, "0")}`] = {
```

Comment:
Any special reason to transform `i` in `000i`?
Maybe we can use just `col_${i}` to avoid extra operations?